lib/search-url-builder.ts

```diff
@@ -0,0 +1,11 @@
+import config from './config'
+import {last, first} from 'lodash'
+
+export const buildTitleFromUrl = (url) => {
```

Comment:
Just an initial poke at this idea. It's a giant of statement at the end of the day feels like?

Reply:
Was it important to keep versions of titles or only URLs? Probably wanna version maps of titles/urls then add a version arg:

`buildTitleFromUrl = (url, version) => `

Reply:
It only matters for the URLs as far as search engines are concerned. 

Titles are pretty mutable. 

The URL builder is from search filter/facet set